Sources/Instrumentation/URLSession/URLSessionInstrumentation.swift

```diff
@@ -573,7 +571,13 @@ public class URLSessionInstrumentation {
     }
 
     private func urlSessionTaskWillResume(_ task: URLSessionTask) {
-        let taskId = self.idKeyForTask(task)
+        // AV Asset Tasks cannot be auto instrumented, they dont include request attributes, skip them
+        if let avAssetDownloadTask = NSClassFromString("__NSCFBackgroundAVAssetDownloadTask"),
```

Comment:
should we move this `classFromString` call  to a static const? maybe it's not very expensive, but will this be called every time this method is called? Is it necessary?

Reply:
Done!!